[2025-06-03T06:59:06.726+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: ingest_raw_data_asia.extract_asia_asia_customer_addresses scheduled__2025-06-02T00:00:00+00:00 [queued]>
[2025-06-03T06:59:06.738+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: ingest_raw_data_asia.extract_asia_asia_customer_addresses scheduled__2025-06-02T00:00:00+00:00 [queued]>
[2025-06-03T06:59:06.739+0000] {taskinstance.py:2170} INFO - Starting attempt 2 of 2
[2025-06-03T06:59:06.922+0000] {taskinstance.py:2191} INFO - Executing <Task(BashOperator): extract_asia_asia_customer_addresses> on 2025-06-02 00:00:00+00:00
[2025-06-03T06:59:06.934+0000] {standard_task_runner.py:60} INFO - Started process 252 to run task
[2025-06-03T06:59:06.945+0000] {standard_task_runner.py:87} INFO - Running: ['airflow', 'tasks', 'run', 'ingest_raw_data_asia', 'extract_asia_asia_customer_addresses', 'scheduled__2025-06-02T00:00:00+00:00', '--job-id', '241', '--raw', '--subdir', 'DAGS_FOLDER/asia/landing.py', '--cfg-path', '/tmp/tmp9dkjvzgh']
[2025-06-03T06:59:06.948+0000] {standard_task_runner.py:88} INFO - Job 241: Subtask extract_asia_asia_customer_addresses
[2025-06-03T06:59:07.077+0000] {logging_mixin.py:188} WARNING - /home/airflow/.local/lib/python3.10/site-packages/airflow/settings.py:194 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-06-03T06:59:07.174+0000] {task_command.py:423} INFO - Running <TaskInstance: ingest_raw_data_asia.extract_asia_asia_customer_addresses scheduled__2025-06-02T00:00:00+00:00 [running]> on host 86069ee61878
[2025-06-03T06:59:07.574+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='ingest_raw_data_asia' AIRFLOW_CTX_TASK_ID='extract_asia_asia_customer_addresses' AIRFLOW_CTX_EXECUTION_DATE='2025-06-02T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-06-02T00:00:00+00:00'
[2025-06-03T06:59:07.575+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-06-03T06:59:07.576+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', '\n                spark-submit                     /opt/airflow/scripts/pyspark_jobs/extract_to_parquet.py                     asia.customer_addresses "jdbc:postgresql://retail-postgres:5432/retail_etl" retail-etl retail_etl 2025-06-02\n            ']
[2025-06-03T06:59:07.596+0000] {subprocess.py:86} INFO - Output:
[2025-06-03T07:01:57.047+0000] {subprocess.py:93} INFO - 25/06/03 07:01:56 INFO SparkContext: Running Spark version 3.5.6
[2025-06-03T07:01:57.404+0000] {subprocess.py:93} INFO - 25/06/03 07:01:57 INFO SparkContext: OS info Linux, 6.11.0-26-generic, amd64
[2025-06-03T07:01:57.405+0000] {subprocess.py:93} INFO - 25/06/03 07:01:57 INFO SparkContext: Java version 11.0.24
[2025-06-03T07:01:59.569+0000] {subprocess.py:93} INFO - 25/06/03 07:01:59 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-06-03T07:02:04.931+0000] {subprocess.py:93} INFO - 25/06/03 07:02:04 INFO ResourceUtils: ==============================================================
[2025-06-03T07:02:04.934+0000] {subprocess.py:93} INFO - 25/06/03 07:02:04 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-06-03T07:02:04.935+0000] {subprocess.py:93} INFO - 25/06/03 07:02:04 INFO ResourceUtils: ==============================================================
[2025-06-03T07:02:04.937+0000] {subprocess.py:93} INFO - 25/06/03 07:02:04 INFO SparkContext: Submitted application: Extract_asia_asia.customer_addresses
[2025-06-03T07:02:05.669+0000] {subprocess.py:93} INFO - 25/06/03 07:02:05 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-06-03T07:02:05.708+0000] {subprocess.py:93} INFO - 25/06/03 07:02:05 INFO ResourceProfile: Limiting resource is cpu
[2025-06-03T07:02:05.711+0000] {subprocess.py:93} INFO - 25/06/03 07:02:05 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-06-03T07:02:08.694+0000] {subprocess.py:93} INFO - 25/06/03 07:02:08 INFO SecurityManager: Changing view acls to: airflow
[2025-06-03T07:02:08.695+0000] {subprocess.py:93} INFO - 25/06/03 07:02:08 INFO SecurityManager: Changing modify acls to: airflow
[2025-06-03T07:02:08.695+0000] {subprocess.py:93} INFO - 25/06/03 07:02:08 INFO SecurityManager: Changing view acls groups to:
[2025-06-03T07:02:08.696+0000] {subprocess.py:93} INFO - 25/06/03 07:02:08 INFO SecurityManager: Changing modify acls groups to:
[2025-06-03T07:02:08.697+0000] {subprocess.py:93} INFO - 25/06/03 07:02:08 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: airflow; groups with view permissions: EMPTY; users with modify permissions: airflow; groups with modify permissions: EMPTY
[2025-06-03T07:02:13.466+0000] {subprocess.py:93} INFO - 25/06/03 07:02:13 INFO Utils: Successfully started service 'sparkDriver' on port 35441.
[2025-06-03T07:02:14.262+0000] {subprocess.py:93} INFO - 25/06/03 07:02:14 INFO SparkEnv: Registering MapOutputTracker
[2025-06-03T07:02:14.741+0000] {subprocess.py:93} INFO - 25/06/03 07:02:14 INFO SparkEnv: Registering BlockManagerMaster
[2025-06-03T07:02:15.140+0000] {subprocess.py:93} INFO - 25/06/03 07:02:15 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-06-03T07:02:15.141+0000] {subprocess.py:93} INFO - 25/06/03 07:02:15 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-06-03T07:02:15.174+0000] {subprocess.py:93} INFO - 25/06/03 07:02:15 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-06-03T07:02:15.681+0000] {subprocess.py:93} INFO - 25/06/03 07:02:15 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-cca63517-c79c-4e62-8a39-1c82da1f78e0
[2025-06-03T07:02:16.040+0000] {subprocess.py:93} INFO - 25/06/03 07:02:16 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-06-03T07:02:16.221+0000] {subprocess.py:93} INFO - 25/06/03 07:02:16 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-06-03T07:02:18.758+0000] {subprocess.py:93} INFO - 25/06/03 07:02:18 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2025-06-03T07:02:19.682+0000] {subprocess.py:93} INFO - 25/06/03 07:02:19 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
[2025-06-03T07:02:19.728+0000] {subprocess.py:93} INFO - 25/06/03 07:02:19 INFO Utils: Successfully started service 'SparkUI' on port 4041.
[2025-06-03T07:02:20.428+0000] {subprocess.py:93} INFO - 25/06/03 07:02:20 INFO Executor: Starting executor ID driver on host 86069ee61878
[2025-06-03T07:02:20.428+0000] {subprocess.py:93} INFO - 25/06/03 07:02:20 INFO Executor: OS info Linux, 6.11.0-26-generic, amd64
[2025-06-03T07:02:20.430+0000] {subprocess.py:93} INFO - 25/06/03 07:02:20 INFO Executor: Java version 11.0.24
[2025-06-03T07:02:20.446+0000] {subprocess.py:93} INFO - 25/06/03 07:02:20 INFO Executor: Starting executor with user classpath (userClassPathFirst = false): ''
[2025-06-03T07:02:20.448+0000] {subprocess.py:93} INFO - 25/06/03 07:02:20 INFO Executor: Created or updated repl class loader org.apache.spark.util.MutableURLClassLoader@64102125 for default.
[2025-06-03T07:02:20.988+0000] {subprocess.py:93} INFO - 25/06/03 07:02:20 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 45009.
[2025-06-03T07:02:20.988+0000] {subprocess.py:93} INFO - 25/06/03 07:02:20 INFO NettyBlockTransferService: Server created on 86069ee61878:45009
[2025-06-03T07:02:21.021+0000] {subprocess.py:93} INFO - 25/06/03 07:02:21 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-06-03T07:02:21.341+0000] {subprocess.py:93} INFO - 25/06/03 07:02:21 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 86069ee61878, 45009, None)
[2025-06-03T07:02:21.370+0000] {subprocess.py:93} INFO - 25/06/03 07:02:21 INFO BlockManagerMasterEndpoint: Registering block manager 86069ee61878:45009 with 434.4 MiB RAM, BlockManagerId(driver, 86069ee61878, 45009, None)
[2025-06-03T07:02:21.383+0000] {subprocess.py:93} INFO - 25/06/03 07:02:21 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 86069ee61878, 45009, None)
[2025-06-03T07:02:21.392+0000] {subprocess.py:93} INFO - 25/06/03 07:02:21 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 86069ee61878, 45009, None)
[2025-06-03T07:02:30.254+0000] {subprocess.py:93} INFO - 25/06/03 07:02:30 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-06-03T07:02:30.376+0000] {subprocess.py:93} INFO - 25/06/03 07:02:30 INFO SharedState: Warehouse path is 'file:/tmp/airflowtmpymi600fe/spark-warehouse'.
[2025-06-03T07:03:29.211+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-06-03T07:03:29.211+0000] {subprocess.py:93} INFO -   File "/opt/airflow/scripts/pyspark_jobs/extract_to_parquet.py", line 24, in <module>
[2025-06-03T07:03:29.212+0000] {subprocess.py:93} INFO -     .load()
[2025-06-03T07:03:29.300+0000] {subprocess.py:93} INFO -   File "/usr/local/spark/python/lib/pyspark.zip/pyspark/sql/readwriter.py", line 314, in load
[2025-06-03T07:03:29.351+0000] {subprocess.py:93} INFO -   File "/usr/local/spark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2025-06-03T07:03:29.351+0000] {subprocess.py:93} INFO -   File "/usr/local/spark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 179, in deco
[2025-06-03T07:03:29.351+0000] {subprocess.py:93} INFO -   File "/usr/local/spark/python/lib/py4j-0.10.9.7-src.zip/py4j/protocol.py", line 326, in get_return_value
[2025-06-03T07:03:29.671+0000] {subprocess.py:93} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling o30.load.
[2025-06-03T07:03:29.672+0000] {subprocess.py:93} INFO - : java.lang.ClassNotFoundException: org.postgresql.Driver
[2025-06-03T07:03:29.672+0000] {subprocess.py:93} INFO - 	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:476)
[2025-06-03T07:03:29.673+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:594)
[2025-06-03T07:03:29.673+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:527)
[2025-06-03T07:03:29.673+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.DriverRegistry$.register(DriverRegistry.scala:46)
[2025-06-03T07:03:29.673+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JDBCOptions.$anonfun$driverClass$1(JDBCOptions.scala:103)
[2025-06-03T07:03:29.673+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JDBCOptions.$anonfun$driverClass$1$adapted(JDBCOptions.scala:103)
[2025-06-03T07:03:29.673+0000] {subprocess.py:93} INFO - 	at scala.Option.foreach(Option.scala:407)
[2025-06-03T07:03:29.674+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JDBCOptions.<init>(JDBCOptions.scala:103)
[2025-06-03T07:03:29.674+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JDBCOptions.<init>(JDBCOptions.scala:41)
[2025-06-03T07:03:29.674+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.jdbc.JdbcRelationProvider.createRelation(JdbcRelationProvider.scala:34)
[2025-06-03T07:03:29.674+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:346)
[2025-06-03T07:03:29.674+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:229)
[2025-06-03T07:03:29.674+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:211)
[2025-06-03T07:03:29.675+0000] {subprocess.py:93} INFO - 	at scala.Option.getOrElse(Option.scala:189)
[2025-06-03T07:03:29.675+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:211)
[2025-06-03T07:03:29.675+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:172)
[2025-06-03T07:03:29.675+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-06-03T07:03:29.675+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2025-06-03T07:03:29.675+0000] {subprocess.py:93} INFO - 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-06-03T07:03:29.676+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[2025-06-03T07:03:29.676+0000] {subprocess.py:93} INFO - 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2025-06-03T07:03:29.676+0000] {subprocess.py:93} INFO - 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2025-06-03T07:03:29.676+0000] {subprocess.py:93} INFO - 	at py4j.Gateway.invoke(Gateway.java:282)
[2025-06-03T07:03:29.676+0000] {subprocess.py:93} INFO - 	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2025-06-03T07:03:29.676+0000] {subprocess.py:93} INFO - 	at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2025-06-03T07:03:29.677+0000] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-06-03T07:03:29.677+0000] {subprocess.py:93} INFO - 	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-06-03T07:03:29.677+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2025-06-03T07:03:29.677+0000] {subprocess.py:93} INFO - 
[2025-06-03T07:03:33.209+0000] {subprocess.py:93} INFO - 25/06/03 07:03:33 INFO SparkContext: Invoking stop() from shutdown hook
[2025-06-03T07:03:33.210+0000] {subprocess.py:93} INFO - 25/06/03 07:03:33 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2025-06-03T07:03:33.285+0000] {subprocess.py:93} INFO - 25/06/03 07:03:33 INFO SparkUI: Stopped Spark web UI at http://86069ee61878:4041
[2025-06-03T07:03:33.707+0000] {subprocess.py:93} INFO - 25/06/03 07:03:33 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2025-06-03T07:03:35.147+0000] {subprocess.py:93} INFO - 25/06/03 07:03:35 INFO MemoryStore: MemoryStore cleared
[2025-06-03T07:03:35.147+0000] {subprocess.py:93} INFO - 25/06/03 07:03:35 INFO BlockManager: BlockManager stopped
[2025-06-03T07:03:35.395+0000] {subprocess.py:93} INFO - 25/06/03 07:03:35 INFO BlockManagerMaster: BlockManagerMaster stopped
[2025-06-03T07:03:35.414+0000] {subprocess.py:93} INFO - 25/06/03 07:03:35 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2025-06-03T07:03:35.523+0000] {subprocess.py:93} INFO - 25/06/03 07:03:35 INFO SparkContext: Successfully stopped SparkContext
[2025-06-03T07:03:35.524+0000] {subprocess.py:93} INFO - 25/06/03 07:03:35 INFO ShutdownHookManager: Shutdown hook called
[2025-06-03T07:03:35.526+0000] {subprocess.py:93} INFO - 25/06/03 07:03:35 INFO ShutdownHookManager: Deleting directory /tmp/spark-f6c1d778-25d0-48b6-ad37-ae6cbaf802e7/pyspark-12da0ea0-59cf-42a6-a4d0-bcc4cfafb967
[2025-06-03T07:03:35.540+0000] {subprocess.py:93} INFO - 25/06/03 07:03:35 INFO ShutdownHookManager: Deleting directory /tmp/spark-3a6327c6-27fc-4590-9fca-590c84bce22a
[2025-06-03T07:03:35.550+0000] {subprocess.py:93} INFO - 25/06/03 07:03:35 INFO ShutdownHookManager: Deleting directory /tmp/spark-f6c1d778-25d0-48b6-ad37-ae6cbaf802e7
[2025-06-03T07:03:36.553+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-06-03T07:04:28.833+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/bash.py", line 212, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-06-03T07:04:45.479+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=ingest_raw_data_asia, task_id=extract_asia_asia_customer_addresses, execution_date=20250602T000000, start_date=20250603T065906, end_date=20250603T070434
[2025-06-03T07:05:28.709+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 241 for task extract_asia_asia_customer_addresses (Bash command failed. The command returned a non-zero exit code 1.; 252)
[2025-06-03T07:05:38.962+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-06-03T07:09:39.587+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
